website/docs/docs/on-premises/maintenance/database.md

```diff
@@ -0,0 +1,262 @@
+
+Notes:
+
+* When you deploy an application to production, the relevant pacticipant version needs to be recorded as deployed in Pactflow, so you need to ensure that you keep any version that you're likely to deploy (or rollback to). Specify a max_age value that is at minimum the number of days it takes between a commit being created and that commit being deployed (with a very comfortable margin of error) and any branch that you deploy from. A reasonable max_age value might be 90 days for the `main` branch. eg. `{"max_age": 90, "mainBranch": true }`
+* If an application is not under active development, a selector that keeps versions by age limit might not actually select any versions. To ensure that we don't lose those critical "latest" versions for our main line of development or our deployed environments, add a selector with `{"mainBranch": true, "latest": true }`, or keep the latest version from each branch by specifying `{"branch": true, "latest": true }`.
+
+### Execution
+
+The database clean tool comes built into the Pactflow On-Premises Docker image. The Docker container should be executed in a compute environment such as AWS Batch. It should run on a regular schedule (eg. daily) and be configured to delete at least as many application versions as are expected to be created between each clean execution.
+
+To execute the clean task, run the Docker container with the appropriate environment variables (documented below), the Pactflow license mounted, and the entrypoint set to `db-clean`.
+
+eg.
+```sh
+docker run --rm \
+           --entrypoint db-clean \
+           --volume ./pactflow-onprem.lic:/home/pactflow-onprem.lic \
+           --env PACTFLOW_DATABASE_URL="..." \ # appropriate environment variables here
+           quay.io/pactflow/enterprise
+```
+
+This example Docker Compose file shows an example configuration. It can be run by saving the contents as `docker-compose.yml` and then running `docker compose up` in the same directory. Note that a Pactflow license file is required to run the Pactflow container.
+
+There will be no data in the database to delete, but the logs will show that the clean process has run.
+
+```yml
+version: "3"
+
+services:
+  postgres:
+    image: postgres:13-alpine
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: password
+      POSTGRES_DB: postgres
+
+  clean:
+    image: quay.io/pactflow/enterprise
+    depends_on:
+      - postgres
+    environment:
+      PACTFLOW_DATABASE_URL: "postgres://postgres:password@postgres/postgres"
+      PACTFLOW_DATABASE_SSLMODE: "disable"
+      PACTFLOW_DATABASE_CLEAN_DELETION_LIMIT: "500"
+      PACTFLOW_SQL_LOG_WARN_DURATION: "60"
+      # Keep the latest version for every pacticipant branch, and all versions less than 30 days old, and anything deployed or released
+      PACTFLOW_DATABASE_CLEAN_KEEP_VERSION_SELECTORS: "[{ \"branch\": true, \"latest\": true }, {\"max_age\": 30 }, {\"deployed\": true }, {\"released\": true }]"
```

Comment:
Is there a way to check that these selectors are valid in advance? I'm assuming the startup logs will indicate if they are borked?

Reply:
Good question - there's actually no validation performed on the selectors at all.